esmvaltool/diag_scripts/weighting/weighted_temperature_map.py

```diff
@@ -61,9 +61,10 @@ def set_antimeridian(dataarray, to: str):
 def mapplot(dataarray, cfg, title_pattern, filename_part, ancestors,
             **colormesh_args):
     """Visualize weighted temperature."""
-    period = '{start_year}-{end_year}'.format(**read_metadata(cfg)['tas'][0])
-    if 'tas_reference' in read_metadata(cfg).keys():
-        meta = read_metadata(cfg)['tas_reference']
+    period = '{start_year}-{end_year}'.format(
+        **read_metadata(cfg)['tas_CLIM_future'][0])
```

Comment:
I'm not a huge fan of this one-liner, do you think it would be possible to define the start/end year before this statement?

```suggestion
    metadata = read_metadata(cfg)
    metadata_future = metadata['tas_CLIM_future']
    
    start_year = metadata_future[0]['start_year']
    end_year = metadata_future[0]['end_year']

    period = f'{start_year}-{end_year}'
```